src/PointwiseFunctions/Hydro/EquationsOfState/PiecewisePolytropicFluid.hpp

```diff
@@ -0,0 +1,174 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <boost/preprocessor/arithmetic/dec.hpp>
+#include <boost/preprocessor/arithmetic/inc.hpp>
+#include <boost/preprocessor/control/expr_iif.hpp>
+#include <boost/preprocessor/list/adt.hpp>
+#include <boost/preprocessor/repetition/for.hpp>
+#include <boost/preprocessor/repetition/repeat.hpp>
+#include <boost/preprocessor/tuple/to_list.hpp>
+#include <limits>
+#include <pup.h>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"  // IWYU pragma: keep
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+/// \endcond
+
+// IWYU pragma: no_forward_declare Tensor
+
+namespace EquationsOfState {
+/*!
+ * \ingroup EquationsOfStateGroup
+ * \brief Equation of state for a piecewise polytropic fluid
+ *
+ * A piecewise polytropic equation of state \f$p=K_i\rho^{\Gamma_i}\f$ where
+ *  \f$K_i\f$ is the polytropic constant and \f$\Gamma_i\f$ is the polytropic
+ * exponent. Here the subscript \f$i\f$ indicates two pairs of constants and
+ *  exponents which characterize `the stiffness' of the matter at low and high
+ *  densities.  For a given density, the polytropic exponent is related to the
+ *  polytropic index \f$N_p\f$ by \f$N_p=1/(\Gamma-1)\f$.
```

Comment:
Maybe add a not that in the future the EOS could be extended to have more piecewise parts? I'm not sure how many pieces BNS simulations use nowadays. I think this would essentially require promoting the `lo_` and `hi_` parts to `std::vector<double>`, one for each segment.